### PR TITLE
[WIP: Do merge] Improve how Preferences are handled

### PR DESCRIPTION
**Pull request by hXtreme**
Refactor out all preferences to preference helper.

Initial Commit for Preferences

+ Add dependency:
    - kotlin
    - preferences

+ Add files
    - PreferenceKeys
    - PreferencesHelper

### CODE REVIEW

**Comment by Doomsdayrs** [comment]
What is the main issue with preferences?

**Comment by hXtreme** [comment]
It is scattered throughout SettingsController this aims to bring all of them under PrefrencesHelper so that SettingsController only has to mange things that are in settings:

eg. Consider togglePause() in SettingsController it is there to manage an option that is not even in the settings.

At the same time, DownloadFragment has to import SettingsController to manage the pause button.

**Comment by Doomsdayrs** [comment]
I see, i did make a mess of those underlying functions

**Comment by hXtreme** [comment]
My soul cried when I saw isReaderLightMode() it's still crying but I'm trying to fix it.

**Comment by Doomsdayrs** [comment]
wait what was so horrible about it?!

**Comment by hXtreme** [comment]
Commits [1c08c41](./26/commits/1c08c4148fdf5d2acd8366c1858498698b9b76a5) and [a1e537a](./26/commits/a1e537a67a8eb3c523eb3da6138097fa06b63ca4) 

Build up to guidelines in [Kodein](https://kodein.org/Kodein-DI/?6.3/getting-started#_definition)

**Comment by hXtreme** [comment]
Oof I shouldn't have rebased it.

File: app/src/main/java/com/github/doomsdayrs/apps/shosetsu/backend/preference/PreferenceKeys.kt [hunk 1]
```diff
@@ -1,5 +1,24 @@
 package com.github.doomsdayrs.apps.shosetsu.backend.preference
 
+/*
```
Comment (hXtreme) [inline app/src/main/java/com/github/doomsdayrs/apps/shosetsu/backend/preference/PreferenceKeys.kt]: Wrong Disclaimer

**Pull request merged**